modules/core/ante/ante_test.go

```diff
@@ -488,3 +489,95 @@ func (suite *AnteTestSuite) TestAnteDecorator() {
 		})
 	}
 }
+
+func (suite *AnteTestSuite) TestAnteDecoratorReCheckTx() {
+	testCases := []struct {
+		name     string
+		malleate func(suite *AnteTestSuite) []sdk.Msg
+		expPass  bool
```

Comment:
This should also be made `expError` with https://github.com/cosmos/ibc-go/issues/6301